src/services/weatherService.js

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+const baseUrl = 'https://api.open-meteo.com/v1';
+
+class WeatherService {
+  async getWeather(latitude, longitude) {
+    try {
+      const params = {
+        latitude,
+        longitude,
+        current:
+          'temperature_2m,apparent_temperature,precipitation,weather_code,wind_speed_10m,wind_direction_10m,wind_gusts_10m',
+      };
+
+      const response = await axios.get(`${baseUrl}/forecast`, { params });
+
+      return response.data;
+    } catch (error) {
+      console.log('Error fetching weather data:', error.message);
+      throw error;
+    }
+  }
+
+  async fetchWeatherData(latitude, longitude) {
```

Comment:
Why do we need to separate `getWeather` and `fetchWeatherData`? It seems can be one method

Reply:
Yeah, you have a point, I'll combine it in one function.